test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -558,6 +776,108 @@ var _ = ginkgo.Describe("Pod controller interacting with scheduler", ginkgo.Orde
+			gomega.Expect(k8sClient.Create(ctx, role2Pod2)).Should(gomega.Succeed())
+		})
+
+		// the composed workload is created
+		wlKey := types.NamespacedName{
+			Namespace: role1Pod1.Namespace,
+			Name:      "dev-pods",
+		}
+		wl := &kueue.Workload{}
+
+		ginkgo.By("checking the composed workload is created", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlKey, wl)).Should(gomega.Succeed())
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		createdPod := &corev1.Pod{}
+		ginkgo.By("check the pods are unsuspended", func() {
+			gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
+				g.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: role1Pod1.Name, Namespace: role1Pod1.Namespace}, createdPod)).
+					To(gomega.Succeed())
+				return createdPod.Spec.SchedulingGates
+			}, util.Timeout, util.Interval).Should(gomega.BeEmpty())
+			gomega.Expect(createdPod.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(spotUntaintedFlavor.Name))
+			gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
+				g.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: role1Pod2.Name, Namespace: role1Pod2.Namespace}, createdPod)).
+					To(gomega.Succeed())
+				return createdPod.Spec.SchedulingGates
+			}, util.Timeout, util.Interval).Should(gomega.BeEmpty())
+			gomega.Expect(createdPod.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(spotUntaintedFlavor.Name))
+			gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
+				g.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: role2Pod1.Name, Namespace: role2Pod1.Namespace}, createdPod)).
+					To(gomega.Succeed())
+				return createdPod.Spec.SchedulingGates
+			}, util.Timeout, util.Interval).Should(gomega.BeEmpty())
+			gomega.Expect(createdPod.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(spotUntaintedFlavor.Name))
+			gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
+				g.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: role2Pod2.Name, Namespace: role2Pod2.Namespace}, createdPod)).
+					To(gomega.Succeed())
+				return createdPod.Spec.SchedulingGates
+			}, util.Timeout, util.Interval).Should(gomega.BeEmpty())
+			gomega.Expect(createdPod.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(spotUntaintedFlavor.Name))
+		})
+		util.ExpectPendingWorkloadsMetric(clusterQueue, 0, 0)
+		util.ExpectReservingActiveWorkloadsMetric(clusterQueue, 1)
+
+		ginkgo.By("deleting the localQueue to prevent readmission", func() {
+			gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, localQueue)).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("clearing the workload's admission to stop the job", func() {
+			gomega.Expect(k8sClient.Get(ctx, wlKey, wl)).Should(gomega.Succeed())
+			gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, wl, nil)).Should(gomega.Succeed())
+			util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, wl)
+		})
```

Comment:
maybe change this to be based on more natural end (set pod phases to Succeeded/ Failed)